scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -764,11 +764,46 @@
                                        {:job-uuid->reserved-host (apply dissoc job-uuid->reserved-host matched-job-uuids)
                                         :launched-job-uuids (into matched-job-uuids launched-job-uuids)})))
 
+(defn distribute-task-requests-to-compute-clusters
+  "Given a collection of un-matched task requests and a collection of
+  compute clusters, distributes the task requests amongst the compute
+  clusters, using a hash of the pending job's uuid."
+  [task-requests compute-clusters]
+  (group-by (fn [{:keys [job]}]
+              (nth compute-clusters
+                   (-> job :job/uuid hash (mod (count compute-clusters)))))
+            task-requests))
+
+(let [pool-name->last-autoscale-atom (atom {})]
```

Comment:
Prefer passing `pool-name->last-autoscale-atom` as an argument, make it easier to test.

Reply:
Nitpick: rename to `pool-name->last-autoscale-time-atom`

Reply:
Will do, thanks

Reply:
Ended up removing this. Since each compute cluster has `:max-pods-outstanding`, and we won't submit more than 1 synthetic pod for each un-matched task, I don't think we need to additionally throttle this based on time.